lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,200 @@ class LogitFormatter(Formatter):
+                # with smallest sum is chosen.
+                diff = np.diff(-np.log(1 / self.locs - 1))
+                space_pessimistic = np.minimum(
+                    np.concatenate(((np.inf,), diff)),
+                    np.concatenate((diff, (np.inf,))),
+                )
+                space_sum = (
+                    np.concatenate(((0,), diff))
+                    + np.concatenate((diff, (0,)))
+                )
+                good_minor = sorted(
+                    range(len(self.locs)),
+                    key=lambda i: (space_pessimistic[i], space_sum[i]),
+                )[-self._minor_number:]
+                self._labelled.update(locs[i] for i in good_minor)
+
+    def _format_value(self, x, locs, sci_notation=True):
+        if sci_notation:
+            exponent = math.floor(np.log10(x))
+            min_precision = 0
+        else:
+            exponent = 0
+            min_precision = 1
+        value = x * 10 ** (-exponent)
+        if len(locs) < 2:
+            precision = min_precision
+        else:
+            diff = np.sort(np.abs(locs - x))[1]
+            precision = -np.log10(diff) + exponent
+            precision = (
+                int(np.round(precision))
+                if is_close_to_int(precision)
+                else math.ceil(precision)
+            )
+            if precision < min_precision:
+                precision = min_precision
+        mantissa = r"%.*f" % (precision, value)
+        if not sci_notation:
+            return mantissa
+        s = r"%s\cdot10^{%d}" % (mantissa, exponent)
         return s
 
+    def _one_minus(self, s):
+        if self._use_overline:
+            return r"\overline{%s}" % s
+        else:
+            return "1-{}".format(s)
+
+    def __call__(self, x, pos=None):
+        if self._minor and x not in self._labelled:
+            return ""
+        if x <= 0 or x >= 1:
+            return ""
+        usetex = rcParams["text.usetex"]
+
+        if is_close_to_int(2 * x) and round(2 * x) == 1:
+            s = self._one_half
+        elif x < 0.5 and is_decade(x, rtol=1e-7):
+            exponent = round(np.log10(x))
+            s = "10^{%d}" % exponent
+        elif x > 0.5 and is_decade(1 - x, rtol=1e-7):
+            exponent = round(np.log10(1 - x))
+            s = self._one_minus("10^{%d}" % exponent)
+        elif x < 0.1:
+            s = self._format_value(x, self.locs)
+        elif x > 0.9:
+            s = self._one_minus(self._format_value(1-x, 1-self.locs))
+        else:
+            s = self._format_value(x, self.locs, sci_notation=False)
+        if usetex:
+            return "$%s$" % s
+        return "$%s$" % _mathdefault(s)
+
     def format_data_short(self, value):
-        """Return a short formatted string representation of a number."""
-        return '%-12g' % value
+        """
+        Return a short formatted string representation of a number.
+        """
+        if value < 0.1:
```

Comment:
Can you add a comment about why these threshold were picked?  It seems odd that we get the `1-N` format only from `0.9 < v < 1`

Reply:
It is only because I dislike format as `1.31e-1' or '1-2.31e-1'. More precisely thresholds choosen to have a exponent less or equal than `-2`.

Comment added.